web/src/lib/components/asset-viewer/photo-sphere-viewer-adapter.svelte

```diff
@@ -28,6 +30,18 @@
       maxFov: 180,
       fisheye: true,
     });
+
+    if (originalImageUrl) {
+      const zoomHandler = ({ zoomLevel }: events.ZoomUpdatedEvent) => {
+        // zoomLevel range: [0, 100]
+        if (Math.round(zoomLevel) >= 75) {
```

Comment:
It might be worth removing this check and loading the original on any zoom event, considering that loading the original can take some time

Reply:
I put in this check mainly to avoid fetching relatively large files (~57 MB original vs 330 kB preview in my case) if the user does not zoom in enough to where the details in the higher resolution original image are significant. The threshold is kind of arbitrary right now though, and user experience will probably vary based on configured preview resolutions.